fmriprep/cli/parser.py

```diff
@@ -0,0 +1,405 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""Parser."""
+import os
+import sys
+from .. import config
+
+
+def _build_parser():
+    """Build parser object."""
+    from functools import partial
+    from pathlib import Path
+    from argparse import (
+        ArgumentParser,
+        ArgumentDefaultsHelpFormatter,
+    )
+    from packaging.version import Version
+    from .version import check_latest, is_flagged
+    from niworkflows.utils.spaces import Reference, SpatialReferences, OutputReferencesAction
+
+    def _path_exists(path, parser):
+        """Ensure a given path exists."""
+        if path is None or not Path(path).exists():
+            raise parser.error("Path does not exist: <%s>." % path)
+        return Path(path).absolute()
+
+    def _min_one(value, parser):
+        """Ensure an argument is not lower than 1."""
+        value = int(value)
+        if value < 1:
+            raise parser.error("Argument can't be less than one.")
+        return value
+
+    def _to_gb(value):
+        scale = {"G": 1, "T": 10**3, "M": 1e-3, "K": 1e-6, "B": 1e-9}
+        digits = ''.join([c for c in value if c.isdigit()])
+        units = value[len(digits):] or "M"
+        return int(digits) * scale[units[0]]
+
+    def _drop_sub(value):
+        value = str(value)
+        return value.lstrip('sub-')
+
+    def _bids_filter(value):
+        from json import loads
+        if value and Path(value).exists():
+            return loads(Path(value).read_text())
+
+    verstr = 'fMRIPrep v{}'.format(config.environment.version)
```

Comment:
Now that we're 3.6+, we should prioritize fstrings IMO

```suggestion
    verstr = f'fMRIPrep v{config.environment.version}'
```

Reply:
Agreed - but in a different, ad-hoc PR.